packages/lesswrong/components/comments/CommentsNewForm.tsx

```diff
@@ -284,8 +298,51 @@ const CommentsNewForm = ({
     if (formDisabledDueToRateLimit) {
       submitBtnProps.disabled = true
     }
-    
+
+    const { DialogueCheckBox } = Components
+
+    const targetUserId = parentComment?.userId
+    const targetUserDisplayName = parentComment?.user?.displayName
+
+    const {
+      userDialogueChecksResult: { results: userDialogueChecks = [] },
+    } = useDialogueMatchmaking({ getMatchedUsers: false, getRecommendedUsers: false, getOptedInUsers: false, getUserDialogueChecks: true });
+
+    const { checkId, userIsChecked, userIsMatched } = parentComment?.user ? getUserCheckInfo(parentComment?.user, userDialogueChecks) : { checkId: null, userIsChecked: null, userIsMatched: null }
+    console.log(document, targetUserId, targetUserDisplayName, checkId, userIsChecked, userIsMatched)
+
+    const { LWHelpIcon, LWTooltip } = Components
+
+    const test = useABTest(showMatchBoxesInCommentReplyFormABTest)
+
+    const topLevelCommentId = parentComment?.topLevelCommentId ?? parentComment?._id
+
+    const { count } = useMulti({
+      terms: {view: "oneReplyInThreadByUser", topLevelCommentId, userId: currentUser?._id},
+      collectionName: "Comments",
+      fragmentName: 'CommentId',
+      skip: !topLevelCommentId || !currentUser,
+    });
+
+    const showDialogueCheckbox = targetUserId && targetUserDisplayName && test === "experiment" && count && count > 0
+
     return <div className={classNames(classes.submit, {[classes.submitMinimalist]: isMinimalist})}>
+    {showDialogueCheckbox ? (
+        <div style={{ marginRight: "auto", display: "inline-block" }}>
+          <div className={ classes.dialogueMatchCheckbox }>
+              <DialogueCheckBox
+                targetUserId={targetUserId}
+                targetUserDisplayName={targetUserDisplayName}
+                checkId={checkId || undefined}
```

Comment:
use `??` rather than `||` unless you specifically want to be either performing the OR operation, or you want the broader clobbering performed by `||`

Reply:
don't understand why one is preferable to the other, but ?? works (which — I read — returns first operand if it's not null), so done